sdk/trace/provider.go

```diff
@@ -210,42 +212,38 @@ func (p *TracerProvider) UnregisterSpanProcessor(sp SpanProcessor) {
 	if p.isShutdown.Load() {
 		return
 	}
-	old := *(p.spanProcessors.Load())
+	old := p.getSpanProcessors()
 	if len(old) == 0 {
 		return
 	}
-	spss := spanProcessorStates{}
-	spss = append(spss, old...)
+	spss := make(spanProcessorStates, len(old))
+	copy(spss, old)
 
 	// stop the span processor if it is started and remove it from the list
-	var stopOnce *spanProcessorState
-	var idx int
 	for i, sps := range spss {
-		if sps.sp == sp {
-			stopOnce = sps
-			idx = i
+		if sps.sp != sp {
+			continue
```

Comment:
It looks like this isn't tested, though it seems it could/should?

Reply:
Added some tests.